frontend/front/src/pages/Public/Components/CanonicalizationLoader.js

```diff
@@ -0,0 +1,60 @@
+import React, { forwardRef, useEffect, useMemo, useState } from "react";
+import { useGetHomeQuery } from "../../../api/apiSlice";
+import Loader from "../../../components/Loader";
+import { Box } from "@mui/material";
+
+const CanonicalizationLoader = forwardRef(({ homeId, onResolved }, ref) => {
+  const [stopPolling, setStopPolling] = useState(false);
+  const {
+    data: validationData,
+    isLoading: isLoadingValidation,
+    isError: validationError,
+  } = useGetHomeQuery(homeId, { pollingInterval: stopPolling ? 0 : 300 });
+
+  const isUncanonicalized = useMemo(
+    () => validationData?.status === "uncanonicalized",
```

Comment:
Put `"uncanonicalized"/"canonicalized"/etc.` in some constants so that nobody can typo the values in the future.